mono/metadata/metadata.c

```diff
@@ -3385,6 +3385,10 @@ do_mono_metadata_parse_type (MonoType *type, MonoImage *m, MonoGenericContainer
 			return FALSE;
 
 		type->data.klass = mono_class_from_mono_type (etype);
+
+		if (transient)
+			g_free (etype);
```

Comment:
Why are you not using mono_metadata_free_type here? Sounds like asking for problems.